awx/main/signals.py

```diff
@@ -561,3 +565,45 @@ def delete_inventory_for_org(sender, instance, **kwargs):
             inventory.schedule_deletion(user_id=getattr(user, 'id', None))
         except RuntimeError, e:
             logger.debug(e)
+
+
+@receiver(post_save, sender=Session)
+def save_user_session_membership(sender, **kwargs):
+    session = kwargs.get('instance', None)
+    if not session:
+        return
+    user = session.get_decoded().get(SESSION_KEY, None)
+    if not user:
+        return
+    user = User.objects.get(pk=user)
+    if UserSessionMembership.objects.filter(user=user, session=session).exists():
+        return
+    UserSessionMembership.objects.create(user=user, session=session, created=timezone.now())
+    for membership in UserSessionMembership.get_memberships_over_limit(user):
+        emit_channel_notification(
+            'control-limit_reached',
+            dict(group_name='control',
+                 reason=unicode(_('limit_reached')),
+                 session_key=membership.session.session_key)
+        )
+
+
+@receiver(post_save, sender=AccessToken)
+def create_access_token_user_if_missing(sender, **kwargs):
+    obj = kwargs['instance']
+    if obj.application and obj.application.user:
+        obj.user = obj.application.user
+        post_save.disconnect(create_access_token_user_if_missing, sender=AccessToken)
+        obj.save()
+        post_save.connect(create_access_token_user_if_missing, sender=AccessToken)
+
+
+@receiver(post_save, sender=User)
+def create_default_oauth_app(sender, **kwargs):
+    if kwargs.get('created', False):
```

Comment:
This looks like it handles newly created users... what do we do for existing users?

Reply:
Good catch. A data migration is probably needed. Ping @rooftopcellist for notice.

Reply:
I don't think there should be a separate application per user at all. An application is typically created only by a user who is configuring the connection between a separate site/app and the OAuth2 provider (AWX).

Reply:
@matburt I remember talking about this and deciding the Default Application is unnecessary.  Is this right, or do we still want it?

Reply:
right, given the work you've done on PATs elsewise.